plugins/signature_provider_plugin/signature_provider_plugin.cpp

```diff
@@ -115,11 +96,30 @@ signature_provider_plugin::signature_provider_for_private_key(const chain::priva
 
 std::optional<std::pair<fc::crypto::blslib::bls_public_key, fc::crypto::blslib::bls_private_key>>
 signature_provider_plugin::bls_public_key_for_specification(const std::string& spec) const {
-   auto [pub_key_str, spec_type_str, spec_data] = my->parse_spec(spec);
+   auto [pub_key_str, spec_type_str, spec_data] = parse_signature_provider_spec(spec);
    if( pub_key_str.starts_with("PUB_BLS") && spec_type_str == "KEY" ) {
       return std::make_pair(fc::crypto::blslib::bls_public_key{pub_key_str}, fc::crypto::blslib::bls_private_key{spec_data});
    }
    return {};
 }
 
+//         public_key   spec_type    spec_data
+std::tuple<std::string, std::string, std::string> signature_provider_plugin::parse_signature_provider_spec(const std::string& spec) {
+   auto delim = spec.find("=");
+   EOS_ASSERT(delim != std::string::npos, chain::plugin_config_exception, "Missing \"=\" in the key spec pair");
+   // public_key can be base64 encoded with trailing `=`
+   // e.g. --signature-provider PUB_BLS_Fmgk<snip>iuA===KEY:PVT_BLS_NZhJ<snip>ZHFu
+   while( spec.size() > delim+1 && spec[delim+1] == '=' )
+      ++delim;
+   EOS_ASSERT(delim < spec.size() + 1, chain::plugin_config_exception, "Missing spec data in the key spec pair");
+   auto pub_key_str = spec.substr(0, delim);
```

Comment:
Although no longer true, PUB_BLS keys `// public_key can be base64 encoded with trailing "="`

Reply:
No need to print out `KEY:`. Just replace it with `***` or `...`.

Reply:
It can print `KEOSD` which could be helpful debug info.

Reply:
> Although no longer true, PUB_BLS keys `// public_key can be base64 encoded with trailing "="`

So we have a loop at lines 112/113 which skips multiple `=` chars. Do we want these included at the end of the pub_key_str?

Reply:
Yes. They are part of the public key. If PUB_BLS key used non-url encoding for the key then it could include a trailing `=`. That trailing `=` is part of the base64 encoded string.